Microsoft.Identity.Web/Client/TokenAcquisition.cs

```diff
@@ -155,110 +137,55 @@ public async Task AddAccountToCacheFromAuthorizationCode(AuthorizationCodeReceiv
 
         /// <summary>
         /// Typically used from an ASP.NET Core Web App or Web API controller, this method gets an access token
-        /// for a downstream API on behalf of the user account which claims are provided in the <see cref="HttpContext.User"/>
-        /// member of the <paramref name="context"/> parameter
+        /// for a downstream API on behalf of the user account
         /// </summary>
         /// <param name="context">HttpContext associated with the Controller or auth operation</param>
         /// <param name="scopes">Scopes to request for the downstream API to call</param>
-        /// <param name="tenantId">Enables to override the tenant/account for the same identity. This is useful in the
-        /// cases where a given account is guest in other tenants, and you want to acquire tokens for a specific tenant</param>
         /// <returns>An access token to call on behalf of the user, the downstream API characterized by its scopes</returns>
-        public async Task<string> GetAccessTokenOnBehalfOfUser(HttpContext context, IEnumerable<string> scopes, string tenant = null)
+        public async Task<string> GetAccessTokenOnBehalfOfUser(HttpContext context, IEnumerable<string> scopes)
         {
             if (context == null)
+            {
                 throw new ArgumentNullException(nameof(context));
+            }
 
             if (scopes == null)
+            {
                 throw new ArgumentNullException(nameof(scopes));
+            }
 
-            // Use MSAL to get the right token to call the API
-            var application = GetOrBuildConfidentialClientApplication(context, context.User);
-
-            // Case of a lazy OBO
-            Claim jwtClaim = context.User.FindFirst("jwt");
-            if (jwtClaim != null)
+            try
             {
-                (context.User.Identity as ClaimsIdentity).RemoveClaim(jwtClaim);
-                var result = await application.AcquireTokenOnBehalfOf(scopes.Except(scopesRequestedByMsalNet), new UserAssertion(jwtClaim.Value))
-                                        .ExecuteAsync();
+                // Use MSAL to get the right token to call the API
+                var application = GetOrBuildConfidentialClientApplication(context, context.User);
+
+                var accounts = await application.GetAccountsAsync();
+
+                var result = await application.AcquireTokenSilent(
+                    scopes.Except(scopesRequestedByMsalNet),
+                    accounts.FirstOrDefault())
+                    .ExecuteAsync();
+
                 return result.AccessToken;
             }
-            else
+            catch (MsalException)
```

Comment:
Actually, no, we should not catch it here, because we want the exceptions to be caught by MsalUiRequiredExceptionFilterAttribute so that it generates a wwwAuthenticate header to leave a chance to the client to setup up in case of MFA or incremental consent.

Reply:
ok. have fixed that

---
In reply to: [318913545](https://github.com/Azure-Samples/active-directory-dotnet-native-aspnetcore-v2/pull/93#discussion_r318913545) [](ancestors = 318913545)